### ISSUE TITLE
Allow to customize caret symbol in `LineEdit`

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#1220]
**Describe the project you are working on:**
A network test

**Describe the problem or limitation you are having in your project:**
there is no way to change the LineEdit control's caret

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
allow us to set a texture for the caret and offset it

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
It would just be a theme and style override

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
Yes and No not that i know if if someone knows you can tell me

**Is there a reason why this should be core and not an add-on in the asset library?:**
Its a pretty basic feature the was overlooked

**Comment by Calinou** [comment]
Can you point to some examples of games that use textured carets?

**Comment by KoBeWi** [comment]
> Can you point to some examples of games that use textured carets?

Judging from "A network test", OP probably tries to create a fancy chat. But this should be said in the proposal. Requesting a feature just because it doesn't exist isn't a valid reason for it to happen.

**Comment by Shadowblitz16** [comment]
@Calinou the issue is not it being textures but letting us style controls how we want.
I just think textures are the best way to style a caret since you can literally make them look however you want.

now that I think of it thought a Style would probable be better suited for this

Most other game frameworks require that you make your own line edits if you need them by drawing the text a caret manually.
Godot doesn't have this because it implements everything as nodes which should cover basic needs like this.

@KoBeWi I agree with you I am suggesting this so I can edit the style of my line edit to fit my font.
this is due to my game being really small resolution 128x128

**Comment by hidemat** [comment]
I my case, I have my resolution set to 1080x1920, and the caret is not visible que the game is run (I think because the with of the caret is to thin).

In this picture the LineEdit node has focus, but I can't see the caret.
![Screenshot 2021-04-23 082943](https://user-images.githubusercontent.com/6559151/115886473-3162fb80-a40e-11eb-94ae-a21a1abbc0d4.png)

Godot ver.: 3.3 stable

**Comment by hidemat** [comment]
@Calinou
base resolution is 1080x1920
test resolution is 360x640

strech>mode  = 2d
strech>aspect = expand
strech>shrink  = 1

**Comment by craigostrin** [comment]
I just wish I could use TextEdit's "block mode" caret. Or a chunky underscore like in Windows cmd:

![image](https://user-images.githubusercontent.com/40326682/134096220-2e71d399-5ad6-4661-8202-0066ec9d984d.png)

**Comment by jabsatz** [comment]
I think this should really be considered, or at least the "block mode" that @craigostrin mentioned. That's very common for arcade style games with a pixely font.

**Comment by Calinou** [comment]
@jabsatz Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Comment by jabsatz** [comment]
wasn't the main issue mentioned here that there weren't examples of games that needed this enough for it to be considered?

**Comment by Shadowblitz16** [comment]
@Calinou can you reopen this I can't seem to edit the offset x/y and size x/y of the caret.
Its not even matching the font..
![image](https://user-images.githubusercontent.com/2453707/173501505-0074fa56-e530-472c-922a-480841e0a798.png)

**Comment by Calinou** [comment]
> @Calinou can you reopen this I can't seem to edit the offset x/y and size x/y of the caret.
> Its not even matching the font..

I don't think we should expose more than the caret width (which is considered to be an accessibility option). If you need more options, set the caret width to 0 and draw a caret yourself.

**Comment by Shadowblitz16** [comment]
@Calinou again running into issues with godot engine when it comes to flexability and dev's thinking they know more then the user.
![image](https://github.com/user-attachments/assets/9a9f01bc-660c-48a6-ac7b-832d78e8b2b2)

![image](https://github.com/user-attachments/assets/26201b97-255c-4eb0-a925-f6b8e56011bc)
